include/tvm/tir/expr.h

```diff
@@ -236,21 +228,33 @@ enum IterVarType : int {
  * \brief Iteration Variable,
  *  represents an iteration over an integer interval.
  */
-class IterVar : public ObjectRef {
+class IterVar : public Var {
  public:
   // construct a new iter var without a domain
   IterVar() {}
   // construct from shared ptr.
-  explicit IterVar(ObjectPtr<Object> n) : ObjectRef(n) {}
+  explicit IterVar(ObjectPtr<Object> n) : Var(n) {}
+  /*! \brief constructor.
+   * \param dom interval of the variable.
+   * \param iter_type indicate the iteration type of the variable.
+   * \param name_hint variable name
+   * \param t data type
+   * \param thread_tag additional tag on the iteration variable.
+   */
+  TVM_DLL IterVar(Range dom, IterVarType iter_type,
```

Comment:
let us cross reference python API to check if it is the right order

Reply:
in python it's `_IterVar(dom, name, iter_type, thread_tag='')`, while if we need to make `name` optional, we have to make `name` after `iter_type`, right?

Reply:
I feel in this case perhaps it is also worth to consider consistency withVar, which means we should put name first. But provide auxilary factory functions(e.g. thread_axis) to quickly construct some of the data